2021-05-05-#17526.md

```diff
@@ -0,0 +1,84 @@
+  an additional fallback Coin Selection strategy. Single Random Draw is fairly straightforward: it
+randomly picks `OutputGroup`s from a pool of eligible UTXOs until the total amount is sufficient to
+cover the payment and fees. Any extra funds are put in a change output.
+
+* This means Coin Selection will have three different solvers: Branch and Bound (BnB), Knapsack, and
+  Single Random Draw (SRD). Note that some randomness is used in Coin Selection, so we won't always
+come up with the same solution.
+
+* The overall strategy within `SelectCoins()` (including PR #17331, on which PR #17526 is built)
+
+	- Coins manually selected by the user using `CoinControl` are selected first.
+
+	- All available coins (excluding the pre-selected ones) are gathered.
+	  `CoinEligibilityFilter`s are used to filter these coins within `SelectCoinsMinConf()`.
+ We have a clear hierarchy of which Coin Selection solutions are preferred: we first try with a
+restriction of at least 6 confirmations on foreign UTXOs and 1 confirmation on our own UTXOs.
+If no solution is found, we try with at least 1 confirmation on all UTXOs. If that doesn't work and
+the user allows spending unconfirmed change, we try that (still requiring at least 1 confirmation on
+foreign UTXOs), gradually increasing mempool ancestor limits on the unconfirmed change.
+
+	- Within `SelectCoinsMinConf()`, `OutputGroup`s are created using the
+	  `CoinEligibilityFilter`. There is a clear preference for BnB and we will only try Knapsack
+and SRD if that fails. However, we'll try both Knapsack and SRD together, picking the solution with
+
+
+## Questions
+
+1. Can you give a high-level description of the Coin Selection strategy proposed in this PR?
+
+2. Within `SelectCoinsMinConf()`, if we have a Knapsack and SRD solution, how do we decide which one
+   to use?
+
+3. Why might we prefer to spend more inputs in the same transaction?
+
+4. Quiz: Based on the Coin Selection scheme proposed here, let's say that Solutions A, B, and C
+   exist (ignore the fact that we would exit early after finding a solution we're satisfied with). Which would we pick?
+
+	 Solution A: picked using Knapsack. Produces a change output, pays 100 satoshis in fees, and
+only uses confirmed UTXOs, each with 4 confirmations.
+
+	Solution B: picked using BnB. No change output, pays 95 satoshis in fees, and uses one
+unconfirmed change output.
+
+	Solution C: picked using SRD. Produces a change output, pays 99 satoshis in fees, and only uses
+confirmed UTXOs, each with 1 confirmation.
+
+
+5. What are `OutputGroups`? Why does SRD pick from output groups rather than from UTXOs?
+
+6. What does calling `GroupOutputs` with `positive_only=true` do (Hint: you may
+   want to review what effective values are)? What could happen if
+`SelectCoinsSRD()` was called with `all_groups` instead of `positive_groups`?
+
+7. What are some ways a deterministic Coin Selection algorithm might leak
```

Comment:
Maybe useful for the notes: https://bitcoin.stackexchange.com/a/32445/5406

Reply:
Thanks! Added in the notes